File: app/src/main/java/me/pwcong/jpstart/mvp/model/PuzzleActivityModelImpl.java
package me.pwcong.jpstart.mvp.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import me.pwcong.jpstart.App;
import me.pwcong.jpstart.R;
import me.pwcong.jpstart.manager.DBManager;
import me.pwcong.jpstart.mvp.bean.JPItem;
import me.pwcong.jpstart.utils.ResourceUtils;

/**
 * Created by Pwcong on 2016/10/24.
 */

public class PuzzleActivityModelImpl implements BaseModel.PuzzleActivityModel {
    @Override
    public String[] getOptions() {
        return new String[]{
                ResourceUtils.getString(App.getInstance(), R.string.hiragana_rome),
                ResourceUtils.getString(App.getInstance(), R.string.hiragana_katakana),
                ResourceUtils.getString(App.getInstance(), R.string.katakana_rome)
        };
    }

    @Override
    public List<JPItem> getItems() {

        List<JPItem> items = new ArrayList<>();
        items.addAll(DBManager.getInstance().getQingYinWithoutHeader());

        Collections.shuffle(items);

        return items;
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/mvp/model/impl/PuzzleActivityModelImpl.kt
package com.github.pwcong.jpstart.mvp.model.impl

import com.github.pwcong.jpstart.App
import com.github.pwcong.jpstart.R
import com.github.pwcong.jpstart.manager.DBManager
import com.github.pwcong.jpstart.mvp.bean.JPItem
import com.github.pwcong.jpstart.mvp.model.BaseModel.PuzzleActivityModel
import com.github.pwcong.jpstart.utils.ResourceUtils

class PuzzleActivityModelImpl() : PuzzleActivityModel {
    override val options: Array<String>
        get() = arrayOf<String>(
            ResourceUtils.getString(App.getInstance(), R.string.hiragana_rome),
            ResourceUtils.getString(App.getInstance(), R.string.hiragana_katakana),
            ResourceUtils.getString(App.getInstance(), R.string.katakana_rome)
        )

    override val items: List<JPItem>
        get() {
            val items: MutableList<JPItem> = mutableListOf()
            items.addAll(DBManager.getInstance().getQingYinWithoutHeader())
            items.shuffle()

            return items
        }
}
